### PR TITLE
feat: improve handling of LOADING response

### PR DESCRIPTION
**Pull request by nesty92**
When a Redis node is loading data (e.g. after restart), it responds with LOADING errors. This makes the client more resilient during Redis node restarts and initial data loading phases.

Close: #656

### CODE REVIEW

**Comment by nesty92** [comment]
Hi @rueian, making some progress here and wanted to confirm that I'm heading in the right direction.

I noticed that Ping wasn't marked as read-only. Is there a specific reason for this, or might it just be an oversight? I’ll wait for your feedback on this before continuing with the cluster client.

**Comment by rueian** [comment]
Hi @nesty92,

I think we are on the right track, but I am concerned about marking PING as readonly. It is also not marked as readonly in the source https://github.com/redis/redis/blob/unstable/src/commands/ping.json.

Can we at least do that in a separate PR?

**Comment by nesty92** [comment]
I agree with your concern about marking PING as readonly. I will handle this in a separate PR to keep the changes isolated and easier to review.

**Comment by nesty92** [comment]
@rueian, I raised [this question](https://github.com/redis/redis/issues/13629) to get feedback from the Redis team regarding why the PING command is not marked as readonly

**Comment by nesty92** [comment]
Hi @rueian, I removed the changes for making `PING` read-only, but handling it in some way is important since `PING` should be the most reliable way to detect the [loading state](https://github.com/redis/docs/blob/37f2a2472cfba1b7ac2d6575de81f53909a68c81/content/commands/ping/index.md?plain=1#L41).

File: client.go [hunk 1]
```diff
@@ -312,7 +312,13 @@ func (c *dedicatedSingleClient) release() {
...
 
 func (c *singleClient) isRetryable(err error, ctx context.Context) bool {
-	return err != nil && err != ErrDoCacheAborted && atomic.LoadUint32(&c.stop) == 0 && ctx.Err() == nil
+	if err == nil || err == ErrDoCacheAborted || atomic.LoadUint32(&c.stop) != 0 || ctx.Err() != nil {
+		return false
+	}
+	if redisErr, ok := err.(*RedisError); ok {
+		return redisErr.IsLoading()
+	}
+	return true
```
Comment (rueian) [inline client.go:321]: Hi @nesty92,

Everything looks good to me, except that the existing rueidiscompat tests are not passing, even after several retries. I am afraid there are something we missed caused by the changes here.

File: client.go [hunk 2]
```diff
@@ -312,7 +312,13 @@ func (c *dedicatedSingleClient) release() {
...
 }
 
 func isRetryable(err error, w wire, ctx context.Context) bool {
```
Comment (rueian) [inline client.go:324]: It turns out this `isRetryable` used by `dedicatedSingleClient` should also be updated.

**Pull request merged**